pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -368,4 +368,22 @@
      * then the ack timeout will be set to 30000 millisecond
      */
     ConsumerBuilder<T> deadLetterPolicy(DeadLetterPolicy deadLetterPolicy);
+
+    /**
+     * If enabled, the consumer will auto subscribe for partitions increasement.
+     * This is only for partitioned consumer and multi-topics consumer.
+     *
+     * @param autoUpdate
+     *            whether to auto update partition increasement
+     */
+    ConsumerBuilder<T> trackPartitionUpdate(boolean autoUpdate);
+
+    /**
+     * Set topics auto update period when setting {@link #trackPartitionUpdate(boolean)} to true for topics consumer.
+     * The period is in minute, and default and minimum value is 1 minute.
+     *
+     * @param periodInMinutes
+     *            number of minutes between checks for topic partitions update
+     */
+    ConsumerBuilder<T> partitionAutoUpdatePeriod(int periodInMinutes);
```

Comment:
I'd leave this config option out, at least for now, for a couple of reasons: 

 * When partitions are increased, the subscriptions are already pre-created on the new partitions. This means that no data will be lost on even though the client will discover the new partitions, even with a greater delay.
 * We should be moving towards using system topics to handle all these broker->client notifications. There's some design work already in progress. Once we have that, then we won't need a refresher timeout.